agent/consul/connect_ca_endpoint.go

```diff
@@ -480,6 +428,30 @@ func (s *ConnectCA) Sign(
 			return fmt.Errorf("SPIFFE ID in CSR from a different trust domain: %s, "+
 				"we are %s", serviceID.Host, signingID.Host())
 		}
+	} else {
```

Comment:
This could also live in the AutoEncrypt.Sign endpoint, then we wouldn't have to check the cert type. And it would only fix it for the first time when we are starting an agent.

Reply:
I wonder if we should enforce the trustdomain for agent certs too, after it was fixed once.

Reply:
The reason why it cannot go there easily is that we would have to move out all the CSR and URI parsing too.